cpp/src/stream_compaction/distinct.cu

```diff
@@ -146,7 +146,7 @@ std::unique_ptr<table> distinct(table_view const& input,
   }
 
   auto const gather_map =
-    get_distinct_indices(input.select(keys), keep, nulls_equal, nans_equal, stream);
+    get_distinct_indices(input.select(keys), keep, nulls_equal, nans_equal, stream, mr);
```

Comment:
The `gather_map` is not returned to the caller and so should use `rmm::mr::get_current_device_resource()` instead of `mr`